src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -142,6 +144,10 @@ def register_options(cls, register):
                   'This is unset by default, because it is generally a good precaution to cache '
                   'only clean/cold builds.')
 
+    register('--native-image-agent-library-dir', type=dir_option, default=None,
```

Comment:
The use of this option does not look specific to `native-image`... it looks like it should work for any Java native agent. If so, should rename.